test/e2e/infra/vagrant/playbook/roles/common/tasks/kube.yml

```diff
@@ -1,15 +1,16 @@
 - name: Add an apt signing key for Kubernetes
   apt_key:
-    url: https://packages.cloud.google.com/apt/doc/apt-key.gpg
+    url: https://pkgs.k8s.io/core:/stable:/v1.28/deb/Release.key
+    keyring: /etc/apt/keyrings/kubernetes-apt-keyring.gpg
     state: present
 
 - name: Adding apt repository for Kubernetes
   apt_repository:
     # kubernetes-xenial should work for Ubuntu 16.04+, there is no
     # kubernetes-bionic
-    repo: deb https://packages.cloud.google.com/apt/ kubernetes-xenial main
+    repo: deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.28/deb/ /
     state: present
-    filename: kubernetes.list
+    filename: /etc/apt/sources.list.d/kubernetes.list
```

Comment:
just `filename: "kubernetes"` would also work.

Ansible will take care of the path and filename extension.

Reply:
Reverted the change.